app/pages/GuaranteeReferenceNumberPage.scala

```diff
@@ -16,11 +16,21 @@
 
 package pages
 
+import models.UserAnswers
 import play.api.libs.json.JsPath
 
+import scala.util.Try
+
 case object GuaranteeReferenceNumberPage extends QuestionPage[String] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "guaranteeReferenceNumber"
+
+  override def cleanup(value: Option[String], userAnswers: UserAnswers, hasChanged: Boolean): Try[UserAnswers] =
+    if (hasChanged) {
+      userAnswers.remove(BalanceIdPage)
+    } else {
+      super.cleanup(value, userAnswers, hasChanged)
+    }
```

Comment:
Very minor refactoring opportunity but could move this to its own trait that extends QuestionPage[String] then have the three page objects extend it

Reply:
I like that, good idea

Reply:
Done, although not totally sure about the name I've given it